app/objects/secondclass/c_fact.py

```diff
@@ -92,7 +92,7 @@ def escaped(self, executor):
 
     def __eq__(self, other):
         if isinstance(other, Fact):
-            return self.unique == other.unique
+            return self.unique == other.unique and self.source == other.source
```

Comment:
if a fact is not equivalent to another fact if the source is different, should that be something we put in `self.unique`?
Or do we use unique elsewhere to grab out all facts that are the same (searching the knowledge store based on `unique`)?

Reply:
I agree, but based on some conversations I've had with various users over the past few months, I don't think that's the only overhaul we should put into unique. As a result, I opted to add source here to validate things in the mean time until we can scope out exactly what unique should be made from.

Reply:
ok, and that sounds good to me, this is already a large PR so I am fine with smaller post cleanup PRs if we can record some of these future changes